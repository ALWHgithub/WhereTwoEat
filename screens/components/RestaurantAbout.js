import React from "react";
import { View, Text, Image, TouchableOpacity, Linking, StyleSheet } from "react-native";
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons";



export default function RestaurantAbout(props) {
  
  const { restaurant, name, image, price, reviews, rating, categories, alias } =
    props.route.params;

  const formattedCategories = categories.map((cat) => cat.title).join(" • ");

  const description = `${formattedCategories} ${
    price ? " • " + price : ""
  } • ${rating} ★ (${reviews}+)`;

  function address(){
    let restAdd = ''
    let data = restaurant.location.display_address;

    for(let i = 0; i < data.length; i++) {
      if(i != 0){
        restAdd += ' '
      }
      restAdd += data[i]
    }
    console.log(restAdd)
    return restAdd
  }
    
  const renderLocation = (location) => {
		if(location == ''){
			return 
		} else {
			return <TouchableOpacity style ={{ flexDirection : "row"}} onPress={() => {Linking.openURL(`https://www.google.com/maps/search/?api=1&query=`+location) }}>
			<MaterialCommunityIcons  name = 'map-marker' size ={25} color = 'grey'/>          
			<Text style={{fontSize: 15, color: 'grey',}}> {location} </Text>
	         </TouchableOpacity>
		}

	}
    
  
  return (
    <View style={styles.container}>
      {/* <Image source={{ uri: 'https://s3-media4.fl.yelpcdn.com/bphoto/ZMV_6YVtVn3jCFpIIiHLqw/o.jpg' }} style={{ width: "100%", height: 180 }} /> */}
      {/* <RestaurantImage image={image} /> */}
      {/* <RestaurantName name={image} /> */}
      <RestaurantName name={name} />
      <RestaurantDescription description={description} />
      {renderLocation(address())}
     
    </View>
  );
}

const RestaurantImage = (props) => (
  <Image source={{ uri: props.image }} style={{ width: "100%", height: 180 }} />
);

const RestaurantName = (props) => (
  <Text
    style={{
      fontSize: 29,
      fontWeight: "600",
      // marginTop: 10,
      // marginHorizontal: 15,
    }}
  >
    {props.name}
  </Text>
);

const RestaurantDescription = (props) => (
  <Text
    style={{
      // marginTop: 10,
      // marginHorizontal: 15,
      fontWeight: "400",
      fontSize: 15,
    }}
  >
    {props.description}
  </Text>
);


const styles = StyleSheet.create({
  container: {
    marginTop: 10,
    marginBottom: 20,
    marginHorizontal: 15,
  },
});