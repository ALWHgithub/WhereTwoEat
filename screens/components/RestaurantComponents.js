import { Text} from 'react-native';


export function renderRating(rating)  {
    if(rating ==1 || rating == 2 || rating == 3 || rating == 4 || rating == 5) {
        return <Text style={{fontSize: 15}} >{rating}.0 </Text> 
    } else {
        return <Text style={{fontSize: 15}} >{rating}</Text> 
    }
  }


export function renderRestrauntsText(restaurantData) {
    if(restaurantData.length == 0) {
      return <Text>Sorry, it dosen't seem like there are any restaurants</Text>
    } else {
      return <Text>Here are the Restaurants !</Text>
    }
  }


