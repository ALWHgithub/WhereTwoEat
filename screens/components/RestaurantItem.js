import React, { useEffect, useState } from "react";
import { render } from "react-dom";
import { View,Text,Image,TouchableOpacity, Dimensions,Linking,use } from 'react-native';
import { useIsFocused } from '@react-navigation/native';
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons"
import config from '../../config'
import StdButton from './button';
import { authentication } from "../../firebase/firebase-config";
import {
  getFirestore,collection,getDocs,
  addDoc, updateDoc, setDoc, doc
} from 'firebase/firestore'

export const localRestaurants = [

]

export default function RestaurantItem(props) {
	const isFocused = useIsFocused();
	const [review, setReview] = useState('')
	const [state, setState] = useState(0)
	const db = getFirestore()
	const colRef = collection(db,'Users')

	  useEffect(() => {
		isFocused && setState(state+1)
	  },[isFocused]);
	
	const toReviews = (restaurant,navigation) => {
		console.log(restaurant.alias)
		navigation.navigate('Reviews',{
			restaurant: restaurant,
			name: restaurant.name,
			image: restaurant.image_url,
			price: restaurant.price,
			reviews: restaurant.review_count,
			rating: restaurant.rating,
			categories: restaurant.categories,
			alias: restaurant.alias,
		})
	}

	
	const renderFavIcon = (id) => {
	if(global.fav == undefined){
		return <TouchableOpacity style={{position: 'absolute', right: 20, top: 20}}  activeOpacity={1.0}>
		<MaterialCommunityIcons name = 'refresh' size ={25} color = "#FFFFFF"/>
		</TouchableOpacity>
	} else if(global.fav != undefined && global.fav.find((res) => (res == id)) != undefined) {
		return <TouchableOpacity style={{position: 'absolute', right: 20, top: 20}} onPress={() => {removeFromFav(id)}} activeOpacity={1.0}>
		<MaterialCommunityIcons name = 'heart' size ={25} color = "#FFFFFF"/>
		</TouchableOpacity>
	} else {
	    return <TouchableOpacity style={{position: 'absolute', right: 20, top: 20}} onPress={() => {addToFav(id)}} activeOpacity={1.0}>
		<MaterialCommunityIcons name = 'heart-outline' size ={25} color = "#FFFFFF"/>
		</TouchableOpacity>
	  }
	}
	
	const addToFav = (id) => {
	    let current = global.fav
	    current.push(id)
	    global.fav = current
		setState(state+1)
		updateDoc(doc(db,'Users',global.user.uid),{fav: current})
	}
	
	const removeFromFav = (id) => {
		let current = global.fav
		current = current.filter(data => data != id);
		global.fav = current
		setState(state+1)
		updateDoc(doc(db,'Users',global.user.uid),{fav: current})
		
	}
	
	const RestaurantInfo = (props) => (
		<View style = {{ flexDirection : "row", 
			justifyContent : "space-between", 
			alignItems : "center",
			marginTop: 10,
		}}>
	
			<View>
			<Text style={{fontSize: 20, textTransform: 'capitalize', fontWeight: 'bold', width: windowWidth - 50}}>
				{/* {replaceAll(props.alias,"-", " ")}  */}
				{props.name}
				</Text>  
				<TouchableOpacity style ={{ flexDirection : "row"}} onPress={() => {Linking.openURL(`tel:${props.phone}`)}}>
				<MaterialCommunityIcons  name = 'phone' size ={15} color = 'grey'/>          
				<Text style={{fontSize: 15, color: 'grey',}}> {props.phone}</Text>
				</TouchableOpacity>
				
			</View>
	
			<View style={{
				flexDirection : "row",
				backgroundColor: "#eee",
				height: 34,
				width: 34,
				alignItems: "center",
				justifyContent: "center",
				borderRadius: 17,
			}}
			>  
				<Text style={{fontSize: 15}} >{props.rating}</Text>
				
			</View>
		</View>
	)
	
	const replaceAll = (str, find, replace) => (
		str.replace(new RegExp(find, 'g'), replace)
	)

	return (
	<>
		{props.restaurantData.map((restaurant,index) => (
			<TouchableOpacity
				key={index}
				activeOpacity={1}
				style={{ marginBottom: 25 }}>
				<View>
					<Image source = {{uri : restaurant.image_url,}} style = {{ width: windowWidth - 10, height : 180, borderRadius: 0, resizeMode: 'cover', }}/>
					{renderFavIcon(restaurant.id)}
					<RestaurantInfo alias = {restaurant.alias} rating = {restaurant.rating} phone = {restaurant.phone} name = {restaurant.name}/>
					<TouchableOpacity onPress={() =>toReviews(restaurant, props.navigation)}>
						<Text style={{fontSize: 15, }}>See Reviews</Text>
					</TouchableOpacity>
				</View>
			</TouchableOpacity>
		))}
	</>
	)
}

const windowWidth = Dimensions.get('window').width;
const windowHeight = Dimensions.get('window').height;




export function VertRestaurantItem(props) {
	
	return (
		<>
			{props.restaurantData.map((restaurant,index) => (
				<View key = {index} >
					<VertRestaurantImage image = {restaurant.image_url}/>
				</View>
			))}
		</>
	)
}

const VertRestaurantImage = (props) => (
	<>
		<Image
		    resizeMethod="scale"
			resizeMode="cover"
			defaultSource = {{
				uri : props.image,
			}}

			style = {{ 
				width: windowWidth * 0.3,
				height : windowHeight * 0.5, 
				borderRadius: 20, 
				resizeMode: 'cover', 
				margin: 5,

			}}
		/>

	</>
)

