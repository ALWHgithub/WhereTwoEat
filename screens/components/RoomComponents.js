import {Text,View,StyleSheet} from 'react-native';

export function renderLoading(loading) {
    if(loading){
      return <Text>Loading</Text>
    } else {
      return 
    }
  }

  export const renderCountPrice = (rooms) => {    
    if(rooms == undefined) {
      return <View style = {[styles.tallyBox, {padding: 0, height: 60, justifyContent: 'center', alignItems: 'center', }]}>
      {/* <Text style = {styles.text}>Current Tally: </Text> */}
      
      <View style ={{flexDirection:'row',}}>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.dollarText}>$</Text>
        <Text>-</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.dollarText}>$$</Text>
        <Text>-</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.dollarText}>$$$</Text>
        <Text>-</Text>
      </View>
      <View style = {[styles.dollarBorder, {borderRightWidth: 0}]}>
        <Text style = {styles.dollarText}>$$$$</Text>
        <Text>-</Text>
      </View>

      </View>
      {/* <Text style = {styles.text}>$: {rooms[1]} $$: {rooms[2]} $$$: {rooms[3]} $$$$: {rooms[4]}</Text> */}
      </View>
    } else {
      return <View style = {[styles.tallyBox, {padding: 0, height: 60, justifyContent: 'center', alignItems: 'center',}]}>
      {/* <Text style = {styles.text}>Current Tally: </Text> */}
      
      <View style ={{flexDirection:'row',}}>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.dollarText}>$</Text>
        <Text>{rooms[1]}</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.dollarText}>$$</Text>
        <Text>{rooms[2]}</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.dollarText}>$$$</Text>
        <Text>{rooms[3]}</Text>
      </View>
      <View style = {[styles.dollarBorder, {borderRightWidth: 0}]}>
        <Text style = {styles.dollarText}>$$$$</Text>
        <Text>{rooms[4]}</Text>
      </View>

      </View>
      {/* <Text style = {styles.text}>$: {rooms[1]} $$: {rooms[2]} $$$: {rooms[3]} $$$$: {rooms[4]}</Text> */}
      </View>
    }
  }

  export const renderCountCat = (rooms) => {    
    if(rooms == undefined) {
      return <View style = {[styles.tallyBox, {padding: 0, height: 120, justifyContent: 'center', alignItems: 'center',}]}>   
      <View style ={{flexDirection:'row',}}>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Chinese</Text>
        <Text>-</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Japanese</Text>
        <Text>-</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Italian</Text>
        <Text>-</Text>
      </View>
      <View style = {[styles.dollarBorder, {borderRightWidth: 0}]}>
        <Text style = {styles.catText}>Cafe</Text>
        <Text>-</Text>
      </View>
      
      </View>
      <View style = {{height: 15 , width: 100}}></View>
      <View style ={{flexDirection:'row',}}>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Burgers</Text>
        <Text>-</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Indian</Text>
        <Text>-</Text>
      </View>
      <View style = {[styles.dollarBorder, {borderRightWidth: 0}]}>
        <Text style = {styles.catText}>All</Text>
        <Text>-</Text>
      </View>
      </View>
      </View>
    } else {
      return <View style = {[styles.tallyBox, {padding: 10, height: 120, justifyContent: 'center', alignItems: 'center', }]}>   
      <View style ={{flexDirection:'row',}}>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Chinese</Text>
        <Text>{rooms['Chinese']}</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Japanese</Text>
        <Text>{rooms['Japanese']}</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Italian</Text>
        <Text>{rooms['Italian']}</Text>
      </View>
      <View style = {[styles.dollarBorder, {borderRightWidth: 0}]}>
        <Text style = {styles.catText}>Cafe</Text>
        <Text>{rooms['Cafe']}</Text>
      </View>
      
      </View>
      <View style = {{height: 15 , width: 100}}></View>
      <View style ={{flexDirection:'row',}}>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Burgers</Text>
        <Text>{rooms['Burgers']}</Text>
      </View>
      <View style = {styles.dollarBorder}>
        <Text style = {styles.catText}>Indian</Text>
        <Text>{rooms['Indian']}</Text>
      </View>
      <View style = {[styles.dollarBorder, {borderRightWidth: 0}]}>
        <Text style = {styles.catText}>All</Text>
        <Text>{rooms['All']}</Text>
      </View>
      </View>
      </View>
    }
  }

  

  export const renderCurrentVotePrice = (votePrice) => {
    if (votePrice == 0) {
      return <Text></Text>
    } else {
      return <Text>Current Vote:  {"$".repeat(votePrice)}</Text>
    }
  }

  const styles = StyleSheet.create({
    text: {
      fontSize: 18,      
    },
    dollarText: {
      textAlign: 'center',
      fontSize: 25, 
    },
    tallyBox: {
      // borderWidth: 1, 
      borderRadius: 5, 
      padding: 20,
      margin: 5,
      backgroundColor: '#e9e9e9',
    },
    dollarBorder: {
      width: '23%', 
      borderRightColor: 'white', 
      borderRightWidth: 4, 
      alignItems: 'center'
    },
    catText: {
      textAlign: 'center',
      fontSize: 16, 
    }
  }); 