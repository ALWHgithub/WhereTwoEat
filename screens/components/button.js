import React from 'react';
import { StyleSheet, TouchableOpacity, Text, View } from 'react-native';

export default function StdButton({ text, onPress}) {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.button}>
        <Text style={styles.buttonText}>{text}</Text>
      </View>
    </TouchableOpacity>
  )
}

export function StdButtonBlue({ text, onPress}) {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.blue}>
        <Text style={styles.buttonText}>{text}</Text>
      </View>
    </TouchableOpacity>
  )
}

export const catButton = (bool,name) => {
  if(bool){
    return <StdButtonBlue text = {name}  />
  } else {
    return <StdButton text = {name} />
  }
  
}

const styles = StyleSheet.create({
  button: {
    borderRadius: 50,
    paddingVertical: 14,
    paddingHorizontal: 18,
    backgroundColor: 'orange',
    alignItems: "center",
    justifyContent: 'center',
    marginHorizontal: 5,
    marginVertical: 5,
  },
  blue: {
    borderRadius: 50,
    paddingVertical: 14,
    paddingHorizontal: 18,
    backgroundColor: '#90ee90',
    alignItems: "center",
    justifyContent: 'center',
    marginHorizontal: 5,
    marginVertical: 5,
  },
  buttonText: {
    color: 'white',
    textTransform: 'uppercase',
    fontWeight: 'bold',
    textAlign: 'center',
  }

})
