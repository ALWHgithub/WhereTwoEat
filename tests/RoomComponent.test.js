import React from 'react';
import renderer from 'react-test-renderer';
import {renderLoading,renderCountPrice} from '../screens/components/RoomComponents'
let room = {"1": 0, "2": 0,"3": 0,"4": 1,"4srvQA39Pif9Eipe1ISJovTJ0u63":  [4,"Italian",],"Chinese": 0,"Italian": 1,"Japanese": 0,"Others": 0,"lat": 0,"long": 0,
"name": "Aaron11","num": 0,"term": " ",}


test('test for true ', () => {
    const tree = renderer.create(
        renderLoading(true)
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

test('test for true ', () => {
const tree = renderer.create(
     renderLoading(false)
    ).toJSON();
expect(tree).toMatchSnapshot();
});

test('test for undefined ', () => {
    const tree = renderer.create(
         renderLoading(undefined)
        ).toJSON();
    expect(tree).toMatchSnapshot();
    });

test('test for data ', () => {
        const tree = renderer.create(
             renderLoading(room)
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });
