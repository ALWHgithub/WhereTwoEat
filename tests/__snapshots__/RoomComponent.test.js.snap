// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for data  1`] = `
<Text>
  Loading
</Text>
`;

exports[`test for true  1`] = `
<Text>
  Loading
</Text>
`;

exports[`test for true  2`] = `null`;

exports[`test for undefined  1`] = `null`;
